### PR TITLE
Added missing allocate_kv_cache() call in CausalLM class

### PR DESCRIPTION
**Pull request by yeonsily**
# What does this PR do?
This function call was missing so we get an error if run with --use_kv_cache and --reuse_cache .

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Review by tthakkal — APPROVED** [review]

**Review by tthakkal — CHANGES_REQUESTED** [review]
I tested this and see there is accuracy issue
```
QUANT_CONFIG=./quantization_config/maxabs_measure.json python ../gaudi_spawn.py \
--use_deepspeed --world_size 2 \
run_generation.py --model_name_or_path google/gemma-7b --use_hpu_graphs --batch_size 1  --bf16 --use_kv_cache --reuse_cache

QUANT_CONFIG=./quantization_config/maxabs_quant.json python ../gaudi_spawn.py \
--use_deepspeed --world_size 2 \
run_generation.py --model_name_or_path google/gemma-7b --use_hpu_graphs --batch_size 1  --bf16 --use_kv_cache --reuse_cache
```

```
Input/outputs:
input 1: ('DeepSpeed is a machine learning framework',)
output 1: ('DeepSpeed is a machine learning framework that aims to',)
```

**Comment by yeonsily** [comment]
> I tested this and see there is accuracy issue
> 
> ```
> QUANT_CONFIG=./quantization_config/maxabs_measure.json python ../gaudi_spawn.py \
> --use_deepspeed --world_size 2 \
> run_generation.py --model_name_or_path google/gemma-7b --use_hpu_graphs --batch_size 1  --bf16 --use_kv_cache --reuse_cache
> 
> QUANT_CONFIG=./quantization_config/maxabs_quant.json python ../gaudi_spawn.py \
> --use_deepspeed --world_size 2 \
> run_generation.py --model_name_or_path google/gemma-7b --use_hpu_graphs --batch_size 1  --bf16 --use_kv_cache --reuse_cache
> ```
> 
> ```
> Input/outputs:
> input 1: ('DeepSpeed is a machine learning framework',)
> output 1: ('DeepSpeed is a machine learning framework that aims to',)
> ```

I think it's not the regression from this change. I checked v1.13.2 with the same command except --reuse_cache and see the same behavior.

File: optimum/habana/transformers/generation/utils.py [hunk 1]
```diff
@@ -99,14 +99,12 @@
...
     "persimmon",
     "qwen2",
-    "starcoder2",
     "llava",
     "llava_next",
     "stablelm",
     "mamba",
     "deci",
     "qwen2_moe",
-    "gemma",
```
Comment (regisss) [inline optimum/habana/transformers/generation/utils.py:109]: Why removing these models from this list?

Comment (yeonsily) [reply optimum/habana/transformers/generation/utils.py:109]: Those two lines are duplicated. You see that 'startcoder2' is in the list twice. same for 'gemma'
https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/transformers/generation/utils.py#L99

Comment (regisss) [reply optimum/habana/transformers/generation/utils.py:109]: Wow great catch!

**Review by regisss — APPROVED** [review]

**Pull request merged**